### PR TITLE
[VAULT-20073] Docs: update upgrade guide for 1.15 with information on…

### PR DESCRIPTION
**Pull request by kubawi**
… Sentinel RGP group policy application

Upgrade guide update companion to https://github.com/hashicorp/vault/pull/23090 (Ensure Role Governing Policies are only applied down the namespace hierarchy).

[Document deployed by Vercel](https://vault-c7y3c07i0-hashicorp.vercel.app/vault/docs/upgrading/upgrade-to-1.15.x)

### CODE REVIEW

**Review by VioletHynes — APPROVED** [review]
LGTM :)

**Comment by peteski22** [comment]
Do we wait until after release day to PR the change to the navigation and remove `"hidden": true` from the `1.15.x` link found in [docs-nav-data.json](https://github.com/hashicorp/vault/blob/main/website/data/docs-nav-data.json)?

```json
{
        "title": "Upgrade to 1.15.x",
        "path": "upgrading/upgrade-to-1.15.x",
  
},
```

File: website/content/docs/upgrading/upgrade-to-1.15.x.mdx [hunk 1]
```diff
@@ -40,3 +40,11 @@ To continue measuring `vault.rollback.attempts.{MOUNTPOINT}` and
 metrics in the `telemetry` stanza of your Vault configuration with the
 [`add_mount_point_rollback_metrics`](/vault/docs/configuration/telemetry#add_mount_point_rollback_metrics)
 option.
+
+## Application of Sentinel Role Governing Policies (RGPs) via identity groups
+
+As of versions 1.15, 1.14.4, and 1.13.8, [the Sentinel RGPSs derived from membership in identity groups apply
```
Comment (peteski22) [inline website/content/docs/upgrading/upgrade-to-1.15.x.mdx:46]:
```suggestion
As of versions `1.15.0`, `1.14.4`, and `1.13.8`, [the Sentinel RGPSs derived from membership in identity groups apply
```

🤔 up to you if you think `this` looks more obvious or not 😄 

Added the `.0` on the version just to keep it the same as the way its formatted in the [other PR](https://github.com/hashicorp/vault/pull/23292)

**Review by peteski22 — APPROVED** [review]
Used the [PR Review comment key](https://hashicorp.atlassian.net/wiki/spaces/VAULT/pages/2545254986/PR+Review+comment+key)

File: website/content/docs/upgrading/upgrade-to-1.15.x.mdx [hunk 2]
```diff
@@ -40,3 +40,11 @@ To continue measuring `vault.rollback.attempts.{MOUNTPOINT}` and
 metrics in the `telemetry` stanza of your Vault configuration with the
 [`add_mount_point_rollback_metrics`](/vault/docs/configuration/telemetry#add_mount_point_rollback_metrics)
 option.
+
+## Application of Sentinel Role Governing Policies (RGPs) via identity groups
```
Comment (mladlow) [inline website/content/docs/upgrading/upgrade-to-1.15.x.mdx:44]: If you add this content in `website/content/partials`, you'll be able to make it a partial and import it into this upgrade guide and the 1.13 and 1.14 upgrade guides more easily.

File: website/content/docs/upgrading/upgrade-to-1.15.x.mdx (hunk 1 above)
Comment (kubawi) [reply website/content/docs/upgrading/upgrade-to-1.15.x.mdx]: Yeah, seems like a good shout to me. Thanks, Pete!

File: website/content/docs/upgrading/upgrade-to-1.15.x.mdx (hunk 2 above)
Comment (kubawi) [reply website/content/docs/upgrading/upgrade-to-1.15.x.mdx:44]: That's very useful. Thanks, Meggie!

**Review by schavis — APPROVED** [review]

**Comment by schavis** [comment]
@peteski22 I can flip the `hidden` field when I merge the 1.15 release notes :)

**Pull request merged**